src/user-federation/UserFederationSection.tsx

```diff
@@ -1,68 +1,176 @@
 import {
+  AlertVariant,
+  Button,
   Card,
   CardTitle,
+  Divider,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
   Gallery,
+  GalleryItem,
   PageSection,
+  PageSectionVariants,
   Split,
   SplitItem,
-  Text,
   TextContent,
-  TextVariants,
+  Title,
+  TitleSizes,
 } from "@patternfly/react-core";
-import { DatabaseIcon } from "@patternfly/react-icons";
+
+import { UserFederationCard } from "./UserFederationCard";
+import { useAlerts } from "../components/alert/Alerts";
+
+import { DatabaseIcon, ExternalLinkAltIcon } from "@patternfly/react-icons";
 import { useTranslation } from "react-i18next";
-import React from "react";
-import { ViewHeader } from "../components/view-header/ViewHeader";
+import "./user-federation.css";
+
+import { RealmContext } from "../context/realm-context/RealmContext";
+import { HttpClientContext } from "../context/http-service/HttpClientContext";
+import { UserFederationRepresentation } from "./model/userFederation";
 
 export const UserFederationSection = () => {
-  const { t } = useTranslation("user-federation");
-  const linkArgs = {
-    title: t("common:learnMore"),
-    href: "http://google.com",
+  const [userFederations, setUserFederations] = useState<
+    UserFederationRepresentation[]
+  >();
+  const { addAlert } = useAlerts();
+
+  const loader = async () => {
+    const testParams: { [name: string]: string | number } = {
+      parentId: `${realm}`,
+      type: "org.keycloak.storage.UserStorageProvider", // MF note that this is providerType in the output, but API call is still type
+    };
+    const result = await httpClient.doGet<UserFederationRepresentation[]>(
+      `/admin/realms/${realm}/components`,
+      {
+        params: testParams,
+      }
+    );
+    setUserFederations(result.data);
   };
 
+  useEffect(() => {
+    loader();
+  }, []);
+
+  const [providerOpen, isProviderMenuOpen] = useState(false);
+
+  const { t } = useTranslation("userFederation");
+
+  const httpClient = useContext(HttpClientContext)!;
+  const { realm } = useContext(RealmContext);
+
+  const ufAddProviderDropdownItems = [
+    <DropdownItem key="itemLDAP">LDAP</DropdownItem>,
+    <DropdownItem key="itemKerberos">Kerberos</DropdownItem>,
+  ];
+
+  let cards;
+
+  if (userFederations) {
+    cards = userFederations.map((userFederation, index) => {
+      const ufCardDropdownItems = [
+        <DropdownItem
+          key={`${index}-cardDelete`}
+          onClick={() => {
+            try {
+              httpClient
+                .doDelete(
+                  `/admin/realms/${realm}/components/${userFederation.id}`
+                )
+                .then(() => loader());
+              addAlert(
+                t("userFederation:userFedDeletedSuccess"),
+                AlertVariant.success
+              );
+            } catch (error) {
+              addAlert(
+                `${t("userFederation:userFedDeleteError")} ${error}`,
```

Comment:
if you change the message bundle to have a param, like this:
```
"userFedDeleteError": "Could not delete user federation: '{{error}}'"
```
then you can change this to:
```suggestion
                t("userFedDeleteError", {error})},
```
also no need to prefix the namespace `userFederation` as that is the default

Reply:
ah, that's nice, i changed it accordingly. also merged my translation strings with existing ones, there were some dups, i temporarily created my own section in the messages to ease the initial merge.